[12.0] add account_asset_management reporting

luc-demeyer opened the pull request:
enhancements compared to 11.0 account_asset_management_xls (https://github.com/OCA/account-financial-tools/pull/701):

- period based reporting (compared to Fiscal Year based in previous versions)
- multiple reporting structures thanks to m2m nature of asset groups
- draft assets can be added to the report
- inconsistent reporting structures are detected

benwillig committed:
[10.0][ADD] Module which provides a fiscal month date range type

benwillig committed:
[10.0][ADD] Module which display the fiscal month on journal entries/items.

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

Commit:
[CHG] prevent to declare fiscal month and/or year for all companies in a multi-company context

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

Commit:
Migrate account_fiscal_month.

Update tests for unlink.

Update tests.

Update unlin ktest in super.

Commit:
Migrate account_move_fiscal_month to 11.0

astirpe committed:
Add button to post all spread lines

astirpe committed:
Improvements as per feedback

astirpe committed:
flake8

astirpe committed:
Better tests

astirpe committed:
Better tests

astirpe committed:
Better tests

astirpe committed:
Handle negative amounts

astirpe committed:
Fix creation of credit invoices

astirpe committed:
Add Dutch translation

astirpe committed:
Update Copyright dates

JordiBForgeFlow committed:
[11.0][ADD] account_move_budget

astirpe committed:
Add option to disable spread planning + code review

astirpe committed:
Add option to force the posting of accounting move

astirpe committed:
Update nl.po

JordiBForgeFlow committed:
[add][account_lock_to_date]

astirpe committed:
Coverage

kittiu committed:
[12.0][ADD] account_payment_netting

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_lock_to_date.pot

astirpe committed:
Code review + add coverage

astirpe committed:
Add partner in move lines

michotm committed:
[FIX] _check_lock_date in account_lock_to_date
if move.company_id.period_lock_to_date or move.company_id.fiscalyear_lock_to_date
are not set nothing should be done.

Saran440 committed:
[12.0][IMP] account_asset_management

oca-travis committed:
[UPD] Update account_move_budget.pot

OCA-git-bot committed:
[UPD] README.rst

AdriaGForgeFlow committed:
[12.0][MIG] account_move_budget

Saran440 committed:
[12.0][FIX] account_asset_management - Allow editing salvage value

OCA-git-bot committed:
[UPD] README.rst

astirpe committed:
Add option to auto-archive spreads

HviorForgeFlow committed:
Update account_move_budget/models/account_move_budget.py

Co-Authored-By: Nikul Chaudhary <nikul.chaudhary.serpentcs@gmail.com>

HviorForgeFlow committed:
Update account_move_budget/models/account_move_budget.py

Co-Authored-By: Nikul Chaudhary <nikul.chaudhary.serpentcs@gmail.com>

luc-demeyer committed:
add account_asset_management reporting

luc-demeyer committed:
add account_asset_management reporting

MiquelRForgeFlow committed:
[MIG] account_fiscal_month: Migration to 12.0

ivantodorovich commented:
Hi @luc-demeyer  :)
thanks for this :)

can you please check travis errors?
there are a few LINT errors and also please add reporting-engine to oca_dependencies so we avoid the missing depenency error

OCA-git-bot committed:
Merge PR #894 into 12.0

Signed-off-by aheficent

oca-travis committed:
[UPD] Update account_fiscal_month.pot

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_invoice_currency
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_invoice_currency/

OCA-git-bot committed:
Merge PR #858 into 12.0

Signed-off-by aheficent

MiquelRForgeFlow committed:
[MIG] account_move_fiscal_month: Migration to 12.0

oca-travis committed:
[UPD] Update account_move_budget.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
Merge PR #743 into 12.0

Signed-off-by aheficent

oca-travis committed:
[UPD] Update account_spread_cost_revenue.pot

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

luc-demeyer committed:
add reporting-engine to oca_depencies

luc-demeyer committed:
add reporting-engine to oca_depencies

luc-demeyer committed:
fix flake8

luc-demeyer committed:
fix flake8

svalaeys committed:
Added translation using Weblate (French)

svalaeys committed:
Translated using Weblate (French)

Currently translated at 100.0% (71 of 71 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_budget
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_budget/fr/

ivantodorovich commented:
Awesome @luc-demeyer ! :)
Now that runbot is building:

**Functional review:**

1. Asset group

Assets Group is required on the wizard, but it's not required on the asset itself. So it's impossible to report assets that has no group.

Maybe the asset group filter should be optional, don't you think?

![image](https://user-images.githubusercontent.com/1914185/66997339-8c59bb00-f0d2-11e9-9f1a-33e67d2f18eb.png)
![image](https://user-images.githubusercontent.com/1914185/66997379-9f6c8b00-f0d2-11e9-8bc4-41e1593524e6.png)

---

2. The company_id field should be shown only if the user has the multi-company group.

![image](https://user-images.githubusercontent.com/1914185/66997871-8b755900-f0d3-11e9-985b-902a9240b9ac.png)

---

3. Report menu location

I think it makes more sense to move it to _Reporting > Assets > Financial Report_
The trend in odoo is to group all the reports in one menu

![image](https://user-images.githubusercontent.com/1914185/66997445-c88d1b80-f0d2-11e9-8b01-da52fcf4bcc0.png)

ivantodorovich commented:
Also this would be great: https://github.com/OCA/account-financial-tools/issues/878

Saran440 committed:
[ADD] options 'use_leap_years'

Commit:
[ADD] allows setting a depreciation ending date

Saran440 committed:
Merge pull request #1 from Eficent/12.0-imp-account_asset_management

[12.0[IMP][account_asset_management] allows setting a depreciation ending date

OCA-git-bot committed:
Merge PR #852 into 12.0

Signed-off-by jbeficent

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/

Commit:
[12.0][FIX][account_asset_management] local variable 'asset' referenced before assignment

tools/account_asset_management/models/account_asset.py", line 970, in _compute_entries
    % (asset.name, asset.code) or asset.name
UnboundLocalError: local variable 'asset' referenced before assignment

pedrobaeza committed:
[MIG] account_asset_management: Handle account_asset_disposal migration

Galoperd committed:
Added translation using Weblate (Spanish)

Galoperd committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (2 of 2 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-base_vat_optional_vies
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-base_vat_optional_vies/es/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 30.5% (36 of 118 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_coa_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_coa_menu/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_position_vat_check
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_position_vat_check/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 91.7% (22 of 24 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_netting
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_netting/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (4 of 4 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_fiscal_year
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_fiscal_year/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_year
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_year/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_month
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_month/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_invoice_constraint_chronology
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_invoice_constraint_chronology/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (7 of 7 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_invoice_currency
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_invoice_currency/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 94.1% (16 of 17 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_lock_date_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_lock_date_update/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_tag_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_tag_menu/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (2 of 2 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-base_vat_optional_vies
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-base_vat_optional_vies/hr/

OCA-git-bot committed:
Merge PR #911 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[ADD] .editorconfig [ci skip]

OCA-git-bot committed:
Merge PR #896 into 12.0

Signed-off-by aheficent

oca-travis committed:
[UPD] Update account_move_fiscal_month.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (200 of 200 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt_BR/

MiquelRForgeFlow committed:
[FIX] Remove unneeded and inefficient "store=True"

OCA-git-bot committed:
Merge PR #915 into 12.0

Signed-off-by aheficent

OCA-git-bot committed:
account_move_fiscal_month 12.0.1.1.0

OCA-git-bot committed:
account_move_fiscal_year 12.0.1.1.0

pedrobaeza committed:
[MIG] account_asset_management: Handle account_asset_analytic migration

Only field renaming is needed.

luc-demeyer committed:
[12.0][FIX]account_move_line_tax_editable

OCA-git-bot committed:
Merge PR #916 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
account_move_line_tax_editable 12.0.1.1.0

OCA-git-bot committed:
Merge PR #904 into 12.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 12.0.2.1.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/

pedrobaeza committed:
[FIX] account_asset_management: Fix tests on leap year

It's bad idea to let dates to be variable.

pedrobaeza committed:
[FIX] account_asset_management: flake8

luc-demeyer committed:
[12.0][FIX]traceback when deviating FY

luc-demeyer committed:
change suggested by pedro

OCA-git-bot committed:
Merge PR #924 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
account_asset_management 12.0.2.1.2

alexis-via committed:
Merge pull request #801 from akretion/12-mig-account_move_template

[12.0][MIG] account_move_template

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

oca-travis committed:
[UPD] Update account_move_template.pot

grindtildeath committed:
[IMP] account_move_budget: Add search view on lines

grindtildeath committed:
[IMP] account_move_budget: Add analytic tags on lines

svalaeys committed:
Fixed dynamic build of the name of the view to be called when errors are found

HviorForgeFlow committed:
[12.0] account_financial_tools: Lint fixing

pedrobaeza committed:
[FIX] account_move_template: Fix RST

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
Merge PR #927 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
account_asset_management 12.0.2.1.3

OCA-git-bot committed:
Merge PR #930 into 12.0

Signed-off-by JordiBForgeFlow

oca-travis committed:
[UPD] Update account_move_template.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_template
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_template/

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Spanish)

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (1 of 1 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_type_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_type_menu/ca/

CarlesAntoli committed:
Added translation using Weblate (Spanish)

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (1 of 1 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_type_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_type_menu/es/

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 92.0% (184 of 200 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/es/

pedrobaeza committed:
[FIX] account_chart_update: Clear caches before searching

If not, changes done in for example taxes are not reflected.

oca-travis committed:
[UPD] Update account_chart_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/

Commit:
[IMP] account_asset_management: Remove asset name limit size

luc-demeyer committed:
[12.0]degressive 4 digits

OCA-git-bot committed:
Merge PR #935 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
account_asset_management 12.0.2.2.0

ivantodorovich commented:
Hi @luc-demeyer 

Do you plan on working on this?
We'd like to work on this to add more information to the report.

We are more than happy to pick up where you left, if you don't have the time. Just let me know :)

luc-demeyer commented:
@ivantodorovich 
I intend to rebase the PR after merge of https://github.com/OCA/account-financial-tools/pull/925
and add a couple of small changes. If you can review PR925 to speed up the merge that would help.

If you have time you can of course always make a PR on top of my PR so that I can review and add it all together. I am interested to know what are the extra informations you intend to add.

ivantodorovich commented:
This is an example report (without sensible data) we're using in a 9.0 project.

[Account.Asset.Report.xlsx](https://github.com/OCA/account-financial-tools/files/4147747/Account.Asset.Report.xlsx)

The original report has 300+ assets.

We'd like to integrate it with this PR (or another PR, after this one is merged).

Some features
- Group by category, with subtotals
- Previous depreciation (depreciation until report start date)
- Period depreciation (depreciation during report start/end date)
- Accumulated depreciation 
- Clean look

luc-demeyer commented:
Looking to the information :

 

That’s already present in the report.

As soon as you define your reporting tag structure you get subtotals per reporting tag.

 

Luc

 

From: Iván Todorovich <notifications@github.com> 
Sent: Monday, 3 February 2020 13:04
To: OCA/account-financial-tools <account-financial-tools@noreply.github.com>
Cc: Luc De Meyer <luc.demeyer@noviat.com>; Mention <mention@noreply.github.com>
Subject: Re: [OCA/account-financial-tools] add account_asset_management reporting (#892)

 

This is an example report (without sensible data) we're using in a 9.0 project.

Account.Asset.Report.xlsx <https://github.com/OCA/account-financial-tools/files/4147747/Account.Asset.Report.xlsx> 

The original report has 300+ assets.

We'd like to integrate it with this PR (or another PR, after this one is merged).

Some features

*	Group by category, with subtotals
*	Previous depreciation (depreciation until report start date)
*	Period depreciation (depreciation during report start/end date)
*	Accumulated depreciation
*	Clean look

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/OCA/account-financial-tools/pull/892?email_source=notifications&email_token=AB2ZSQO4QPMI52IP4LRROCDRBAB3DA5CNFSM4I5524K2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEKTTAZA#issuecomment-581382244> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB2ZSQOUWCC727KMV7YQYQLRBAB3DANCNFSM4I5524KQ> .  <https://github.com/notifications/beacon/AB2ZSQODGVBU5BDMIHBPMJDRBAB3DA5CNFSM4I5524K2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEKTTAZA.gif>

ivantodorovich commented:
> As soon as you define your reporting tag structure you get subtotals per reporting tag.

What do you mean? :thinking: 

I feel like I'm missing something huge here :sweat_smile: 

Where are we supposed to configure these tags?


This is the report I get from this module:
[Testinggroup_asset_report.xlsx](https://github.com/OCA/account-financial-tools/files/4148024/Testinggroup_asset_report.xlsx)

luc-demeyer commented:
The tags replace the "parent_id" reporting structure (has been there since version 6 of this module) but this concept is richer since now you can overlay multiple reporting structures (e.g. a legal one and one for your specific business needs). 
The account_asset_management_xls reporting module (which will now be integrated in the base module with this PR) has always given  subtotals.

Concerning reporting structures, have a look at https://github.com/luc-demeyer/noviat-apps/tree/11.0/l10n_be_asset_management_data.
This one adds the legal belgian asset reporting structure. As soon as the new tag based reporting structure is merged I'll publish an update of this localisation module for v12.

ivantodorovich commented:
So, what you mean is that we should create a `l10n_fr_asset_management_data` module, to implement this structure?

And, basically, extend `['account.asset']`: to overload `_xls_acquisition_template`, and other `_xls_*` methods, :thinking: ?

luc-demeyer commented:
@ivantodorovich 
in case there exist a legal french reporting structure for assets than it is indeed recommended to create a small data module to add this structure.
If you want to add extra fields to the standard report than I propose that we extend the _xls_acquisition_template in the base module because I can hardly imagine that your requirement will be limited to France only.
In case the requirement is generic than it would also be best to add the field also to the _xls_acquisition_fields. 
Localisation modules can freely add/remove fields by modifying the _xls_acquisition_fields. That's a five minutes effort.

sbejaoui committed:
[IMP] - use model_create_multi for account move line creation

OCA-git-bot committed:
Merge PR #938 into 12.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 12.0.2.2.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/

OCA-git-bot committed:
Merge PR #859 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
account_asset_management 12.0.2.2.2

OCA-git-bot committed:
[UPD] README.rst

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (63 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/pt/

Martronic-SA committed:
Added translation using Weblate (French)

Martronic-SA committed:
Translated using Weblate (French)

Currently translated at 90.0% (135 of 150 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_spread_cost_revenue
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_spread_cost_revenue/fr/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (17 of 17 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_lock_date_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_lock_date_update/nl/

pedrobaeza committed:
[MIG] account_asset_management: Add asset initial entry

On OCA module, an initial depreciation line is created as summary of the
asset. We recreate that line here for the old assets.

pedrobaeza committed:
[MIG] account_asset_management: Set asset line previous pointer

Set pointer to the previous depreciation line on each asset line.
This must be executed before `add_asset_initial_entry` for being
populated correctly.

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (17 of 17 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_lock_date_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_lock_date_update/ca/

Martronic-SA committed:
Translate wizard choices.

Extra wizard selection ar not translated.

OCA-git-bot committed:
Merge PR #940 into 12.0

Signed-off-by astirpe

oca-travis committed:
[UPD] Update account_spread_cost_revenue.pot

OCA-git-bot committed:
account_spread_cost_revenue 12.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_spread_cost_revenue
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_spread_cost_revenue/

Martronic-SA committed:
Translated using Weblate (French)

Currently translated at 90.2% (138 of 153 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_spread_cost_revenue
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_spread_cost_revenue/fr/

AaronHForgeFlow committed:
[MIG]account_lock_to_date to v12

AaronHForgeFlow committed:
[UPT]account_update_lock_to_date rebranding

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 99.5% (199 of 200 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (200 of 200 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 22.0% (26 of 118 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (69 of 69 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_template
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_template/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_invoice_constraint_chronology
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_invoice_constraint_chronology/pt/

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_year
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_year/ca/

CarlesAntoli committed:
Added translation using Weblate (Spanish)

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_year
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_year/es/

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (27 of 27 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_renumber
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_renumber/ca/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (118 of 118 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (10 of 10 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_partner_required
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_partner_required/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (63 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (34 of 34 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_document_reversal
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_document_reversal/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_month
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_month/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_fiscal_year
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_fiscal_year/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (7 of 7 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_invoice_currency
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_invoice_currency/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (17 of 17 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_lock_date_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_lock_date_update/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (3 of 3 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_menu/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (71 of 71 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_budget
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_budget/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (3 of 3 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_fiscal_month
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_fiscal_month/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 35.3% (54 of 153 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_spread_cost_revenue
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_spread_cost_revenue/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (1 of 1 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_type_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_type_menu/pt/

alvarorib committed:
Added translation using Weblate (Portuguese)

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (2 of 2 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-base_vat_optional_vies
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-base_vat_optional_vies/pt/

pedrobaeza committed:
[MIG+IMP] account_asset_management: Use other column for running migration scripts

Old versions of account_asset module contains a parent_id column for assets, which
triggers this migration scripts, but giving error later, as the expected data is
not there (only account_asset_management module contains it), so we use another
column for detecting the condition.

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 39.0% (78 of 200 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 91.3% (63 of 69 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_template
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_template/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (13 of 13 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_group_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_group_menu/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (3 of 3 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_menu/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (3 of 3 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_fiscal_month
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_fiscal_month/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_line_purchase_info
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_line_purchase_info/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 47.1% (16 of 34 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_document_reversal
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_document_reversal/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 2.8% (2 of 71 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_budget
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_budget/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 36.5% (23 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/hr/

Commit:
[FIX] account_check_deposit: New check group_invoices

OCA-git-bot committed:
Merge PR #950 into 12.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
account_check_deposit 12.0.1.0.1

pedrobaeza committed:
[FIX+IMP] account_chart_update: Create taxes in batch

With this, we gain some performance, but also prevent an error if there are template
taxes that have children taxes, as Odoo standard method forces that in the same call
all of them are created simultaneously, or it will enter in an infinite loop.

pedrobaeza committed:
[FIX+IMP] account_chart_update: Add consistency method

With this, we prevent lock due to infinite loop if children taxes are matched, but
not the parent one and it's marked to be created.

Closes #912

pedrobaeza committed:
[FIX] account_asset_management: Avoid error

Steps to reproduce the problem:

* Go to assets view
* Group by profile
* Unfold a group and click on an asset
* Click on "Journal Entries" smart-button
* Go back to the asset list
* Click again on the same asset (or another).
* Click on "Journal Entries" smart-button

Current behavior:

Error saying "KeyError: 'profile_id'"

Expected behavior:

No error

The cause for this is that Odoo stores in the context the key `group_by` with the
value `profile_id` in the specified chain of steps. That context entry is used for
grouping records in the list, and system tries to group the journal entries also
by that field, which doesn't exists in the other model, and thus the error.

We avoided it copying the context to be passes and leaving out that entry.

pedrobaeza committed:
[IMP] account_asset_management: Simplify code for opening related entries

We don't need the search having the one2many field and no need also for list
comprehension having mapped operator.

oca-travis committed:
[UPD] Update account_asset_management.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/

oca-travis committed:
[UPD] Update account_chart_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/

OCA-git-bot committed:
Merge PR #941 into 12.0

Signed-off-by AaronHForgeFlow

oca-travis committed:
[UPD] Update account_lock_to_date.pot

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
Merge PR #926 into 12.0

Signed-off-by AaronHForgeFlow

oca-travis committed:
[UPD] Update account_move_budget.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_budget
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_budget/

OCA-git-bot committed:
Merge PR #936 into 12.0

Signed-off-by AaronHForgeFlow

pedrobaeza committed:
[IMP] account_chart_update: Search for real XML-IDs only

Previously, only a check on an existing XML-ID which is not coming from exportation
was done, but now we check that the XML-ID is the exact one. With this, we avoid
that the same record changed their XML-ID and thus, there's no more link between
them.

oca-travis committed:
[UPD] Update account_chart_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/

OCA-git-bot committed:
Merge PR #823 into 12.0

Signed-off-by kittiu

oca-travis committed:
[UPD] Update account_payment_netting.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

sysadminmatmoz committed:
Added translation using Weblate (Slovenian)

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_coa_menu
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_coa_menu/sl/

ValentinVinagre committed:
[FIX]account_move_template: fix computed amount

Commit:
Merge pull request #960 from sygel-technology/12.0-fix-account_move_template

[12.0][FIX]account_move_template: fix computed amount

oca-travis committed:
[UPD] Update account_move_template.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_move_template
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_move_template/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (27 of 27 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_renumber
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_renumber/pt/

msdruidoo committed:
Translated using Weblate (French)

Currently translated at 98.4% (62 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/fr/

msdruidoo committed:
Translated using Weblate (French)

Currently translated at 100.0% (63 of 63 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_check_deposit/fr/

luc-demeyer committed:
order, name_get, name_search

luc-demeyer committed:
group optional in xlsx report, usability enhancements

luc-demeyer committed:
fix merge conflicts

luc-demeyer commented:
@ivantodorovich 
I have rebased this PR and pushed an update  PR which implements the suggestions made by you in your review:
- Asset group is now optional in the reporting wizard. A flat report is generated when no group is 
 specified (although I still recommend to add localisation modules with asset reporting structure, cf. https://github.com/Noviat/noviat-apps/tree/12.0/l10n_be_asset_management_data for an example.
- the asset report is now moved under the "reporting" main menu as suggested by you
- added multi-company group on company-id

On top of this I also implemented a couple of small usability enhancements.
Can you review ?

ivantodorovich commented on .editorconfig:
```diff
@@ -0,0 +1,20 @@
+indent_size = unset
+indent_style = unset
+insert_final_newline = false
+trim_trailing_whitespace = false
```
Is this file supposed to be here?

ivantodorovich reviewed (approved):
:+1: looks good! thanks!

luc-demeyer commented:
This one is ready for merge (and port to 13.0)

@pedrobaeza 
Can you merge (ocabot major) and also answer ivantodorovich remarks on the .editorconfig ?
I'll make the 13.0 PR after merge.

pedrobaeza commented:
@carlosdauden can you please review this?

@luc-demeyer is it possible to squash initial bunch of commits from @astirpe ?

qgroulard commented on account_asset_management/models/account_asset_group.py:
```diff
@@ -23,7 +24,42 @@ class AccountAssetGroup(models.Model):
+            if list_view:
+                name = rec.code
+            else:
+                name = rec.code + ' ' + rec.name
```
`code` might be empty and cause an error.

luc-demeyer committed:
fix account.asset.group name_get

luc-demeyer commented:
@qgroulard 
Thanks for your feedback.
Fixed via https://github.com/OCA/account-financial-tools/pull/892/commits/5920f1fd7199d408c95b94e0f06363e4d1c58364

I now truncate the name when there is no code.
The asset list view now gets the following look & feel:

![image](https://user-images.githubusercontent.com/7706945/80619668-4bf5d500-8a45-11ea-80d2-f895e8e6eb38.png)

luc-demeyer commented:
@pedrobaeza 
I don't feel very comfortable to squash a bunch of commits from 2018. 
Can you give me the syntax on how to do it to avoid that I squash too much ?

pedrobaeza commented:
I can do it if you prefer.

luc-demeyer commented:
@pedrobaeza 
as you can imagine I prefer that you squash the historical commits.

pedrobaeza commented:
Hehe, OK, it was just a formality for not doing it unadvertised. I'll do it soon and tell you back. I also pinged my colleague for reviewing this and finally merge it.

carlosdauden reviewed (approved):
Good work

luc-demeyer commented:
@pedrobaeza can you merge ?

pedrobaeza commented:
Fixed the mess in #980, as I can't push to your branch. Please never ever use GitHub conflict resolution nor the equivalent in git through merge operation. It's not first time I have to fix similar things and it requires a lot of time to fix conflicts. You should check the method you use. If you tell me the git used commands, I can tell you where is the problem.

Any way, closing this and continuing in the other, where my main concern is the added dependency.
